### PR TITLE
V2.1 resolve issues

### PR DESCRIPTION
**Pull request by qianz-z**
To fix bugs pointed out in the issues

### CODE REVIEW

File: src/main/java/recipeditor/command/DeleteCommand.java [hunk 1]
```diff
@@ -11,7 +11,7 @@ public class DeleteCommand extends Command {
             + '\n' + "\t /delete -t <recipe_title>";
     private static final String COMMAND_FUNCTION =
             "Delete the recipe of given title or index from recipeditor.";
-    public static final String CORRECT_FORMAT = "The input should be '/delete <flag> <argument>.'";
+    public static final String CORRECT_FORMAT = "The input should be '/delete <recipe flag> <argument>.'";
```
Comment (Brominne) [inline src/main/java/recipeditor/command/DeleteCommand.java:14]: Not very precise to say `<recipe flag>` here, as in `EditCommand`, `<recipe flag>` mean `-t`,`-d`,`-i`,`-s`, but here its only `-id` or `-t`

File: src/main/java/recipeditor/command/ExitCommand.java [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
 public class ExitCommand extends Command {
     public static final String COMMAND_TYPE = "/exit";
     public static final String EXIT_MESSAGE = "RecipEditor ends here...";
-    private static final String COMMAND_SYNTAX = "/exit";
+    public static final String COMMAND_NAME = "exit";
     private static final String COMMAND_FUNCTION = "Exit recipeditor.";
 
     public ExitCommand() {
-        super(COMMAND_SYNTAX, COMMAND_FUNCTION);
+        super(COMMAND_TYPE, COMMAND_FUNCTION);
```
Comment (Brominne) [inline src/main/java/recipeditor/command/ExitCommand.java:12]: similar to in `AddCommand`, `COMMAND_SYNTAX` and `COMMAND_TYPE` are same value but different meaning

File: src/main/java/recipeditor/command/ListCommand.java [hunk 3]
```diff
@@ -6,11 +6,11 @@
 
 public class ListCommand extends Command {
     public static final String COMMAND_TYPE = "/list";
-    private static final String COMMAND_SYNTAX = "/list";
+    public static final String COMMAND_NAME = "list";
     private static final String COMMAND_FUNCTION = "Show a list of all recipe titles in recipeditor.";
 
     public ListCommand() {
-        super(COMMAND_SYNTAX, COMMAND_FUNCTION);
+        super(COMMAND_TYPE, COMMAND_FUNCTION);
```
Comment (Brominne) [inline src/main/java/recipeditor/command/ListCommand.java:13]: same as in `AddCommand` and `ExitCommand`

File: src/main/java/recipeditor/parser/Parser.java [hunk 4]
```diff
@@ -229,26 +251,22 @@ private static Command parseFindCommand(String[] parsed) {
...
-    public static Command parseHelpCommand(String[] parsed) {
-
-        if (parsed.length == COMMAND_INPUT_LENGTH) {
-            logger.log(Level.INFO, "Help command initialised");
-            return new HelpCommand(parsed[1]);
-        }
-        return new InvalidCommand(HelpCommand.CORRECT_FORMAT + HelpCommand.HELP_MESSAGE);
+    public static Command parseHelpCommand(String params) {
+        logger.log(Level.INFO, "Help command initialised");
+        return new HelpCommand(params);
```
Comment (Brominne) [inline src/main/java/recipeditor/parser/Parser.java:256]: the original change account for the case of excess arguments in `HelpCommand`, so you do not want to report issue when `HelpCommand` contains excess arguments?

File: src/main/java/recipeditor/parser/Parser.java [hunk 5]
```diff
@@ -147,7 +169,7 @@ private static Command parseViewCommand(String[] parsed) {
         } catch (IndexOutOfBoundsException i) {
             Ui.showMessage(WRONG_COMMAND_FORMAT_MESSAGE);
         } catch (NumberFormatException n) {
-            Ui.showMessage(parsed[2] + INVALID_INDEX_MESSAGE);
+            Ui.showMessage(InvalidCommand.INVALID_MESSAGE);
```
Comment (Brominne) [inline src/main/java/recipeditor/parser/Parser.java:172]: the original message shows: 
```
<tndex> is not valid index
```
which i think is clearer

**Review by Brominne — CHANGES_REQUESTED** [review]

**Comment by Brominne** [comment]
All comments resolved, can proceed to merge, thank you!

**Comment by qianz-z** [comment]
Thanks for reviewing! @Brominne

**Pull request merged**